src/brsTypes/components/Regex.ts

```diff
@@ -0,0 +1,153 @@
+import { BrsBoolean, BrsInvalid, BrsString, BrsValue, ValueKind } from "../BrsType";
+import { BrsComponent } from "./BrsComponent";
+import { BrsType } from "..";
+import { Callable, StdlibArgument } from "../Callable";
+import { Interpreter } from "../../interpreter";
+import { BrsArray } from "./BrsArray";
+
+export class Regex extends BrsComponent implements BrsValue {
+    readonly kind = ValueKind.Object;
+    private jsRegex : RegExp;
+
+    constructor(expression: BrsString, flags = new BrsString("")) {
+        super("roRegex");
+        this.jsRegex = new RegExp(expression.value, flags.value);
+
+        this.registerMethods([
+            this.isMatch,
+            this.match,
+            this.replace,
+            this.replaceAll,
+            this.split,
+            this.matchAll
+        ]);
+    }
+
+    toString(parent?: BrsType): string {
+        return "<Component: roRegex>";
+    }
+
+    private parseReplacementPattern(pattern: string): string {
+        return pattern.replace(/\\/g, "\$");
+    }
+
+    private isMatch = new Callable(
+        "ismatch",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.Boolean
+            },
+            impl: (interpreter: Interpreter, str: BrsString) => {
+                return this.jsRegex.test(str.value) ? BrsBoolean.True : BrsBoolean.False;
+            }
+        }
+    );
+
+    private match = new Callable(
+        "match",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.Object
+            },
+            impl: (interpreter: Interpreter, str: BrsString) => {
+                const result = this.jsRegex.exec(str.value);
+                let arr = [];
+                if (result !== null) {
+                    for (let i = 0; i < result.length; i++) {
+                        let item = result[i] ? new BrsString(result[i]) : BrsInvalid.Instance;
+                        arr.push(item);
+                    }
+                }
+
+                return new BrsArray(arr);
+            }
+        }
+    );
+
+    private replace = new Callable(
+        "replace",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String),
+                    new StdlibArgument("str", ValueKind.String)
```

Comment:
Would you mind updating this argument to be called `"replacement"` like it is in the reference docs?  These names get reported to users when a type mismatch is encountered, so it'd be nice to keep it clear 😄

Reply:
🤦‍♂️